runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/cache/KafkaCachePartition.java

```diff
@@ -621,6 +630,25 @@ else if (headerTypes != null && !headerTypes.isEmpty())
             }
         }
 
+        if (convertKey != ConverterHandler.NONE &&
+            transforms != null && transforms.extractKey != null)
+        {
+            final ConverterHandler.FieldVisitor writeKey = (buffer, index, length) ->
+            {
+                final int position = entryMark.value + FIELD_OFFSET_PADDED_KEY;
+                KafkaCachePaddedKeyFW paddedKey =
+                    logFile.readBytes(position, paddedKeyRO::wrap);
+                final int paddedKeySize = paddedKey.sizeof();
+                KafkaCachePaddedKeyFW.Builder paddedKeyBuilder = paddedKeyRW;
+                final int keySize = paddedKeyBuilder.key(k -> k.length(length).value(buffer, index, length)).sizeof();
+                paddedKeyBuilder.padding(logFile.buffer(), 0, paddedKeySize - keySize - SIZE_OF_INT);
+                KafkaCachePaddedKeyFW newPaddedKey = paddedKeyBuilder.build();
+                logFile.writeBytes(position, newPaddedKey.buffer(), newPaddedKey.offset(), newPaddedKey.sizeof());
+            };
+
+            convertKey.extracted(transforms.extractKey, writeKey);
+        }
+
```

Comment:
I think this is assuming that `extract-key` is always processed after `extract-key`, which might be fine for now if we cannot extract the key from a header yet, but probably not sufficient for the general case later where these transforms are expected to execute in the order defined, the output of the previous feeding into the input of the next.